worker/src/main/java/gov/cms/ab2d/worker/processor/domainmodel/ProgressTracker.java

```diff
@@ -97,6 +98,21 @@ public int getTotalCount() {
         return totalCount;
     }
 
+    public int getTotalPossibleCount() {
+        int remainingToBeDefined = numContracts - patientsByContracts.size();
+        return getTotalCount() + (remainingToBeDefined * getTotalAverageNumber());
+    }
+
+    public int getTotalAverageNumber() {
+        // If we don't have any contract mappings, assume that it's max value so the denominator will underestimate amount done
+        if (getTotalCount() == 0) {
+            return Integer.MAX_VALUE;
+        }
+        // return the average number of patients in known contracts
+        IntSummaryStatistics stats = patientsByContracts.stream().mapToInt(c -> c.getPatients().size()).summaryStatistics();
```

Comment:
If this is done partway through before all patient counts per contract have been figured out what sort of number will it give?

Reply:
If there are two out of three contracts defined, the potential total number of patients are a + b + average(a + b). If a is 10, b is 20 and we don't know about c, the potential is estimated to be 45.

Reply:
Ok. I guess my concern was since when I saw the issue in prod the progress shot up pretty quickly to 400% done (when it was probably really only 30%) is it now going to go to 99% pretty quickly too with the same sponsor?